pkg/api/v1/status/atlascluster.go

```diff
@@ -72,7 +99,28 @@ func AtlasClusterMongoDBVersionOption(mongoDBVersion string) AtlasClusterStatusO
 
 func AtlasClusterConnectionStringsOption(connectionStrings *mongodbatlas.ConnectionStrings) AtlasClusterStatusOption {
 	return func(s *AtlasClusterStatus) {
-		cs := ConnectionStrings(*connectionStrings)
+		pe := make([]PrivateEndpoint, 0, len(connectionStrings.PrivateEndpoint))
+		for _, v := range connectionStrings.PrivateEndpoint {
+			endpoints := make([]Endpoint, 0, len(v.Endpoints))
+			for _, e := range v.Endpoints {
+				endpoints = append(endpoints, Endpoint(e))
+			}
+
+			pe = append(pe, PrivateEndpoint{
+				ConnectionString:    v.ConnectionString,
+				Endpoints:           endpoints,
+				SRVConnectionString: v.SRVConnectionString,
+				Type:                v.Type,
+			})
+		}
+
+		cs := ConnectionStrings{
```

Comment:
am I right that we do field-by-field building because of the custom `PrivateEndpoint` built above?

Reply:
Also because we're now missing the deprecated `AwsPrivateEndpoint...` fields